apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx

```diff
@@ -257,12 +259,20 @@ export const metadata = {
   description: 'Generated by create next app',
 }
 
-export default function RootLayout({ children }) {
+export default async function RootLayout({ children }) {
+  const supabase = createServerComponentSupabaseClient({
+    headers,
+    cookies,
+  })
+
+  const {
+    data: { session },
+  } = await supabase.auth.getSession()
+
   return (
     <html lang="en">
       <body>
         <SupabaseProvider session={session}>
-          <SupabaseListener serverAccessToken={session?.access_token} />
```

Comment:
The `<SupabaseListener />` component is present in the Supabase auth examples, however it's not really necessary here since the `<SupabaseProvider />` snippet seems to cover the logic that it contains in the auth examples.